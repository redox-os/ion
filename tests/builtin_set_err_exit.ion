set -e

if false
else
    echo "ignore in if condition"
end

test 0 -eq 1 || echo "ignore in test condition"

if not false
    echo "ignore in if not condition"
end

false || echo "ignore in or"

true && false || echo "ignore in and"

if ! matches $(cat /proc/self/stat | cut -d' ' -f1) "fail"
    echo "ignore in matches"
end

let a = 1
while test $a -lt 3
    echo "ignore in while loop condition"
    let a += 1
end
echo "after a while ..."

false | true
echo "ignore pipes with successful last"

# still fails in the normal case
false && true

echo unreachable
