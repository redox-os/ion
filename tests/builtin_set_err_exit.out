ignore in if condition
ignore in test condition
ignore in if not condition
ignore in or
ignore in and
ignore in matches
ignore in while loop condition
ignore in while loop condition
after a while ...
ignore pipes with successful last
ion: pipeline execution error: early exit: pipeline failed with error code 1
